src/contracts/pods/EigenPod.sol

```diff
@@ -354,18 +291,68 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         eigenPodManager.recordBeaconChainETHBalanceUpdate(podOwner, int256(totalAmountToBeRestakedWei));
     }
 
-    /// @notice Called by the pod owner to withdraw the nonBeaconChainETHBalanceWei
-    function withdrawNonBeaconChainETHBalanceWei(
-        address recipient,
-        uint256 amountToWithdraw
-    ) external onlyEigenPodOwner onlyWhenNotPaused(PAUSED_NON_PROOF_WITHDRAWALS) {
+    /**
+     * @dev Prove that one or more validators were slashed on the beacon chain and have not had timely
+     * checkpoint proofs since being slashed. If successful, this allows the caller to start a checkpoint.
+     * @dev Note that in order to start a checkpoint, any existing checkpoint must already be completed!
+     * (See `_startCheckpoint` for details)
+     * @param beaconTimestamp the beacon chain timestamp sent to the 4788 oracle contract. Corresponds
+     * to the parent beacon block root against which the proof is verified.
+     * @param stateRootProof proves a beacon state root against a beacon block root
+     * @param proof the fields of the beacon chain "Validator" container, along with a merkle proof against
+     * the beacon state root. See the consensus specs for more details:
+     * https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#validator
+     *
+     * @dev Staleness conditions:
+     * - Validator's last balance update is older than `beaconTimestamp` by `TIME_TILL_STALE_BALANCE`
+     * - Validator MUST be in `ACTIVE` status in the pod
+     * - Validator MUST be slashed on the beacon chain
+     */
+    function verifyStaleBalance(
+        uint64 beaconTimestamp,
+        BeaconChainProofs.StateRootProof calldata stateRootProof,
+        BeaconChainProofs.ValidatorProof calldata proof
+    )
+        external
+        onlyWhenNotPaused(PAUSED_VERIFY_STALE_BALANCE)
+    {  
+        bytes32 validatorPubkey = proof.validatorFields.getPubkeyHash();
+        ValidatorInfo memory validatorInfo = _validatorPubkeyHashToInfo[validatorPubkey];
+
+        // Validator must be eligible for a staleness proof
         require(
-            amountToWithdraw <= nonBeaconChainETHBalanceWei,
-            "EigenPod.withdrawnonBeaconChainETHBalanceWei: amountToWithdraw is greater than nonBeaconChainETHBalanceWei"
+            beaconTimestamp > validatorInfo.lastCheckpointedAt + TIME_TILL_STALE_BALANCE,
+            "EigenPod.verifyStaleBalance: validator balance is not stale yet"
         );
-        nonBeaconChainETHBalanceWei -= amountToWithdraw;
-        emit NonBeaconChainETHWithdrawn(recipient, amountToWithdraw);
-        _sendETH_AsDelayedWithdrawal(recipient, amountToWithdraw);
+
+        // Validator must be checkpoint-able
+        require(
+            validatorInfo.status == VALIDATOR_STATUS.ACTIVE,
+            "EigenPod.verifyStaleBalance: validator is not active"
+        );
+
+        // Validator must be slashed on the beacon chain
```

Comment:
in the case where there is a huge liveness failure for no slashing this doesnt work...

Reply:
PEPE is not resilient against the beacon chain itself failing, and I think that's a fair compromise. If the beacon chain itself is failing I'm not really even sure how the execution layer copes with that, so... not sure what I'm supposed to do about that.

Ultimately the goal is to do better than this stale balance function post-Pectra, but for now I think this is good enough.

Reply:
I am open to other ideas for `verifyStaleBalance` if you have them, but I don't want to complicate things too much.